.github/workflows/integration.yaml

```diff
@@ -32,6 +33,45 @@ jobs:
       - uses: actions/checkout@v5
         with:
           persist-credentials: false
-      - uses: astral-sh/setup-uv@d0cc045d04ccac9d8b7881df0226f9e82c39688e  # v6.8.0
+      - uses: astral-sh/setup-uv@557e51de59eb14aaaba2ed9621916900a91d50c6  # v6.6.1
       - run: uv tool install tox --with tox-uv
       - run: tox -e integration -- --log-cli-level=INFO -s -k "${{ matrix.test }}"
+
+  secrets:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        juju-channel:
+          - 3.1/stable
+          - 3.2/stable
+          - 3.3/stable
+          - 3.4/stable
+          - 3.5/stable
+          - 3.6/stable
+          # Juju 4.0 beta 6 doesn't support actions.
+          # https://discourse.charmhub.io/t/the-juju-insider-mongoless-beta/18383
+          # - 4.0/beta
+
+    continue-on-error: ${{ matrix.juju-channel == '4.0/beta' }}
+
+    steps:
+      - run: |
+           # Work around https://github.com/jnsgruk/concierge/issues/30
+           sudo apt-get remove -y docker-ce docker-ce-cli containerd.io
+           sudo rm -rf /run/containerd
+        if: matrix.preset == 'k8s'
+      - run: sudo snap install --classic concierge
+      - run: >
+          sudo concierge prepare
+          --juju-channel=${{ matrix.juju-channel }}
+          --charmcraft-channel=3.x/stable
+          -p microk8s
+      - uses: actions/checkout@v5
+        with:
+          persist-credentials: false
+      - uses: astral-sh/setup-uv@557e51de59eb14aaaba2ed9621916900a91d50c6  # v6.6.1
+      - run: uv tool install tox --with tox-uv
+      - run: tox -e integration -- --log-cli-level=INFO -s test/integration/test_secrets.py
```

Comment:
I guess `test_secrets` will be run once by the `test` job above, and again here for the 3/stable (3.6/stable) track. Do we care?